src/Component/Tests/Symfony/EventListener/DeserializeListenerTest.php

```diff
@@ -0,0 +1,133 @@
+        $operation->getDenormalizationContext()->willReturn(null)->shouldBeCalled();
+
+        $request->isMethodSafe()->willReturn(false);
+        $request->getRequestFormat()->willReturn('json');
+        $request->getContent()->willReturn(['food' => 'fighters']);
+
+        $operation->canDeserialize()->willReturn(null)->shouldBeCalled();
+        $operation->getDenormalizationContext()->willReturn([])->shouldBeCalled();
+
+        $this->serializer->deserialize(['food' => 'fighters'], 'App\Resource', 'json', [])->willReturn($data)->shouldBeCalled();
+
+        $this->deserializableListener->onKernelRequest($event->reveal());
+    }
+
+    /** @test */
+    public function it_deserializes_data_with_denormalization_context(): void
+    {
+        $event = $this->prophesize(RequestEvent::class);
+        $request = $this->prophesize(Request::class);
+        $operation = $this->prophesize(HttpOperation::class);
+        $data = $this->prophesize(\stdClass::class);
+
+        $event->getRequest()->willReturn($request);
+
+        $this->operationInitiator->initializeOperation($request)->willReturn($operation);
+
+        $request->attributes = new ParameterBag();
+
+        $request->isMethodSafe()->willReturn(false);
+        $request->getRequestFormat()->willReturn('json');
+        $request->getContent()->willReturn(['food' => 'fighters']);
+
+        $operation->getResource()->willReturn(new ResourceMetadata(alias: 'app.dummy', class: 'App\Resource'));
+        $operation->canDeserialize()->willReturn(null)->shouldBeCalled();
+        $operation->getDenormalizationContext()->willReturn(['groups' => ['dummy:write']]);
+
+        $this->serializer->deserialize(['food' => 'fighters'], 'App\Resource', 'json', ['groups' => ['dummy:write']])->willReturn($data)->shouldBeCalled();
+
+        $this->deserializableListener->onKernelRequest($event->reveal());
+    }
+
+    /** @test */
+    public function it_does_nothing_if_operation_cannot_be_deserialized(): void
+    {
+        $event = $this->prophesize(RequestEvent::class);
+        $request = $this->prophesize(Request::class);
+        $operation = $this->prophesize(HttpOperation::class);
+        $data = $this->prophesize(\stdClass::class);
+
+        $event->getRequest()->willReturn($request);
+
+        $this->operationInitiator->initializeOperation($request)->willReturn($operation);
+
+        $request->attributes = new ParameterBag();
+
+        $request->isMethodSafe()->willReturn(false);
+        $request->getRequestFormat()->willReturn('json');
+        $request->getContent()->willReturn(['food' => 'fighters']);
+
+        $operation->getResource()->willReturn(new ResourceMetadata(alias: 'app.dummy', class: 'App\Resource'));
+        $operation->canDeserialize()->willReturn(false)->shouldBeCalled();
+
+        $this->serializer->deserialize(['food' => 'fighters'], 'App\Resource', 'json', [])->willReturn($data)->shouldNotBeCalled();
```

Comment:
```suggestion
        $this->serializer->deserialize(Argument::cetera())->shouldNotBeCalled();
```
No point in mocking a return or being specific about the arguments since it shouldn't get called in the first place.

Reply:
Hum cause this deserialize is the main part of this class, I think it's better to be specific.

Reply:
@NoResponseMate Sorry my bad, you are right.

Reply:
done

Reply:
You forgot about the `->willReturn` 👋